p2p/peer.go

```diff
@@ -138,10 +140,6 @@ func (pc peerConn) ID() ID {
 
 // Return the IP from the connection RemoteAddr
 func (pc peerConn) RemoteIP() net.IP {
-	if pc.ip != nil {
-		return pc.ip
-	}
```

Comment:
I think the idea of caching the result so we don't need to resolve the IP every time this method is called still makes sense. Perhaps what you need to do when checking `IPHasChanged`, is to set the p.ip to nil before calling `RemoteIP`

Reply:
updated